keps/sig-cluster-lifecycle/wgs/1815-componentconfig-objectmeta/README.md

```diff
@@ -0,0 +1,696 @@
+-->
+
+Existing configurations that lack the ObjectMeta field will continue to work as the objectMeta field will be optional. This should work within the following guidelines:
+
+- Not specified, not defined: Common case today.
+- Specified, but not defined in the type: Some users do this today for interop with tools. Enabling strict decoding breaks this. Thus, we must define it in the type to enable strict decoding.
+- Not specified, but defined: ObjectMeta should be kept optional, so that this remains backwards-compatible with "not specified, not defined".
+- Specified, and defined: Works correctly with strict decoding. We need to add ObjectMeta so that strict decoding remains backwards with "specified, but not defined in the type".
+
+### Version Skew Strategy
+
+<!--
+If applicable, how will the component handle version skew with other
+components? What are the guarantees? Make sure this is in the test plan.
+
+Consider the following in developing a version skew strategy for this
+enhancement:
+- Does this enhancement involve coordinating behavior in the control plane and
+  in the kubelet? How does an n-2 kubelet without this feature available behave
+  when this feature is used?
+- Will any other components on the node change? For example, changes to CSI,
+  CRI or CNI may require updating that component before the kubelet.
+-->
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable, can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/20190731-production-readiness-review-process.md
+
+Production readiness review questionnaire must be completed for features in
+v1.19 or later, but is non-blocking at this time. That is, approval is not
+required in order to be in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+
+-->
+
+### Feature enablement and rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [ ] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name:
+    - Components depending on the feature gate:
+  - [x] Other
+    - Describe the mechanism: Editing the existing component config file to add the new minimum required field (metadata.name) and restarting the component.
```

Comment:
We could probably just say: "Editing the existing component config file that is consumed by tooling to add the new minimum required field (metadata.name). This proposal allows additional schema for tooling compatibility, but this schema does not enable any new features of a given component."